pyflakes/test/test_doctests.py

```diff
@@ -190,6 +190,50 @@ def doctest_stuff():
             '''
         """, m.UndefinedName)
 
+    def test_nested_class(self):
+        """Doctest within nested class are processed."""
+        self.flakes("""
+        class C:
+            class D:
+                '''
+                    >>> m
+                '''
+                def doctest_stuff(self):
+                    '''
+                        >>> m
+                    '''
+                    return 1
+        """, m.UndefinedName, m.UndefinedName)
+
+    def test_ignore_nested_function(self):
+        """Doctest module does not process doctest in nested functions."""
+        # the names inner_function and m generate UndefinedName
+        # if the doctest was processed.
+        self.flakes("""
+        def doctest_stuff():
+            def inner_function():
+                '''
+                    >>> inner_function()
+                    1
+                    >>> m
+                '''
+                return 1
+            m = inner_function()
+            return m
+        """)
```

Comment:
Would a simpler test suffice? For example a would-be SyntaxError:

        self.flakes('''
        def doctest_stuff():
            def inner_function():
                """
                    >>> if True:
                    ... pass
                """
        ''')